pyQuARC/code/custom_validator.py

```diff
@@ -98,6 +100,25 @@ def presence_check(*field_values):
 
         return {"valid": validity, "value": value}
 
+    @staticmethod
+    def granule_sensor_presence_check(sensor_values, collection_shortname=None, version=None, dataset_id=None):
+        """
+        Checks if sensor is provided at the granule level if provided at
+        collection level
+        """
+        if dataset_id == None:
+            collection  = requests.get(f'{CMR_URL}/search/collections.umm_json?short_name={collection_shortname}&version={version}').json()
+        else: 
+            collection  = requests.get(f'{CMR_URL}/search/collections.umm_json?DatasetId={dataset_id}').json()
```

Comment:
Can we refactor this? Form the url first, using if else and then get the response from the url?